frontend/windows/handle_container.go

```diff
@@ -33,81 +41,228 @@ var (
+	})
+
+	getBaseRef := func(idx int, spec *dalec.Spec) string {
+		baseRef := refs[idx]
+
+		updated := dalec.GetBaseOutputImage(spec, targetKey)
+		if updated == "" {
+			return baseRef
 		}
 
-		bc, err := dockerui.NewClient(client)
-		if err != nil {
-			return nil, nil, err
+		if bases.len() == 0 {
+			return updated
 		}
 
-		targetPlatform, err := getTargetPlatform(bc)
+		warnBaseOverride()
+		return baseRef
+	}
+
+	rb, err := dcBuild(ctx, dc, func(ctx context.Context, platform *ocispecs.Platform, idx int) (ref gwclient.Reference, retCfg, retBaseCfg *dalec.DockerImageSpec, retErr error) {
+		spec, err := frontend.LoadSpec(ctx, dc, platform, frontend.WithAllowArgs(
+			argBasesPathKey,
+			argBasesContextKey,
+		))
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
+		}
+
+		if err := validateRuntimeDeps(spec, targetKey); err != nil {
+			return nil, nil, nil, fmt.Errorf("error validating windows spec: %w", err)
 		}
 
 		pg := dalec.ProgressGroup("Build windows container: " + spec.Name)
 		worker, err := distroConfig.Worker(sOpt, pg)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
 
 		bin, err := buildBinaries(ctx, spec, worker, client, sOpt, targetKey)
 		if err != nil {
-			return nil, nil, fmt.Errorf("unable to build binary %w", err)
+			return nil, nil, nil, fmt.Errorf("unable to build binary %w", err)
 		}
 
-		baseImgName := getBaseOutputImage(spec, targetKey, defaultBaseImage)
-		baseImage := llb.Image(baseImgName, llb.Platform(targetPlatform))
+		baseRef := getBaseRef(idx, spec)
+		baseImage := llb.Image(baseRef, llb.Platform(*platform))
 
 		out := baseImage.
 			File(llb.Copy(bin, "/", windowsSystemDir)).
 			With(copySymlinks(spec.GetImagePost(targetKey)))
 
 		def, err := out.Marshal(ctx)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
 
 		res, err := client.Solve(ctx, gwclient.SolveRequest{
 			Definition: def.ToPB(),
 		})
 		if err != nil {
-			return nil, nil, err
-		}
-
-		imgRef := dalec.GetBaseOutputImage(spec, targetKey)
-		if imgRef == "" {
-			imgRef = defaultBaseImage
+			return nil, nil, nil, err
 		}
 
-		_, _, dt, err := client.ResolveImageConfig(ctx, imgRef, sourceresolver.Opt{
-			Platform: &targetPlatform,
-		})
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "could not resolve base image config")
+		var baseCfg dalec.DockerImageSpec
+		if err := json.Unmarshal(cfgs[idx], &baseCfg); err != nil {
+			return nil, nil, nil, errors.Wrap(err, "error unmarshalling base image config")
 		}
 
+		// Get a copy of the cfg so we can modify it
 		var img dalec.DockerImageSpec
-		if err := json.Unmarshal(dt, &img); err != nil {
-			return nil, nil, errors.Wrap(err, "error unmarshalling base image config")
+		if err := json.Unmarshal(cfgs[idx], &img); err != nil {
+			return nil, nil, nil, errors.Wrap(err, "error unmarshalling base image config")
```

Comment:
Are this message and line 245 both true in that they're both `unmarshalling base image config`?